source/blog/2019-03-02-update-on-module-unification-and-octane.md

```diff
@@ -0,0 +1,379 @@
+2. Similarly, ecosystem tools don't understand MU. We have to build custom
+   integrations to get things like "Go to Definition" to work in IDEs or code
+   splitting to work in webpack, that other libraries get for free.
+3. JavaScript files in Ember use native module syntax anyway, which doesn't
+   go through the MU system, adding to the confusion. Having one system in
+   a component's JavaScript and another in its template is not ideal.
+4. When file names are so meaningful, it can be frustrating to figure out how
+   to tell MU ignore a file that _isn't_ meaningful. Tasks that are trivial
+   in other programming environments, such as grouping related components
+   together, can easily turn into a battle where your build starts erroring
+   because you can't figure out the right incantation of the MU rules.
+
+### A Personal Anecdote
+
+Personally, this last one was what really caused me to step back and
+re-evaluate our plan for MU. It happened during a project at work where we
+were using both Glimmer.js (with Module Unification) and Preact.
+
+As the number of components grew, a co-worker created a directory called
+`icons` in the Preact app to hold all of the components for rendering
+different SVG icons. I'm sure it didn't take more than a few minutes to
+create the directory, drag the appropriate component files in, and update the
+paths everywhere those components were imported. (In fact, VS Code probably
+updated the import paths automatically.)
+
+When we tried to do the same thing in the Glimmer app, it was a much
+different experience that turned into an hours-long slog. And despite all the
+great things it _does_ do, MU doesn't really have a way to do this kind of
+lightweight grouping.
+
+We could have extracted the SVG icon components into a separate package. We
+could have tried to create a higher-order component that wrapped all of the
+child icon components. We could have found a workaround. But it seemed
+ridiculous to burn so much time looking for a "workaround" to perform a task
+that felt like it should have been (and was, in Preact) trivial.
+
+I knew, intellectually, the benefits of MU. I knew how carefully it was
+designed to enforce structure and consistency as your application grew and
+had different engineers of different experience levels working on it. (Indeed,
+by the end of the project, I found the Glimmer app much easier to navigate,
+while the Preact app had several inconsistently-followed conventions.)
+
+But I never forgot how viscerally bad it felt to have my co-workers fight so
+hard to do something that felt like it should be so easy.
+
+### The Path Forward
+
+So this was the status quo last year. We were all incredibly frustrated that
+we couldn't make progress on the scoped package problem, but I was even more
+overwhelmed trying to figure out what, if anything, to do about the negative
+experiences my co-workers had had when using MU.
```

Comment:
```suggestion
experiences my co-workers had when using MU.
```

Reply:
This is actually grammatically correct: it's past perfect tense, plus the past participle of _have_.

For example, this sentence in the present perfect:

> They have had a bad experience today.

Becomes this sentence in the past perfect:

> They had had a bad experience yesterday.